gateway/api/management/commands/free_resources.py

```diff
@@ -27,6 +29,14 @@ def handle(self, *args, **options):
 
             # only kill cluster if not in local mode and no jobs are running there
             if len(alive_jobs) == 0 and not settings.RAY_CLUSTER_MODE.get("local"):
+                # Add a delay in killing the ray cluster
+                # default value is 0, meaning no delay
+                if settings.RAY_CLUSTER_SHUTDOWN_DELAY == -1:
+                    # -1 means don't delete the ray cluster at all
+                    return
+                if settings.RAY_CLUSTER_SHUTDOWN_DELAY > 0:
+                    # a value greater than zero means to delay shutting down
+                    time.sleep(settings.RAY_CLUSTER_SHUTDOWN_DELAY)
```

Comment:
This call blocks the processing of the scheduler.  The other jobs are not processed during this sleep.

Reply:
good catch. if this was go, I'd call a goroutine here. guess we need to find the python equivalent... :thinking:

Reply:
or, maybe for now to keep things simple, we make it just a don't delete cluster option and punt the variable delay down the road (most of what we need is probably covered by not deleting...)

Reply:
That's fine with me.  With the not delete, the raycluster CRD instances stay in the cluster.  We may want something like [this](https://github.com/Qiskit-Extensions/quantum-serverless/pull/698).

Reply:
yeah, I think that would be a good thing to include -- a way to clear out all the old clusters...
(thinking out loud) maybe we should include a label when the no-delete option is on, so that we can easily find clusters that won't be cleaned up automatically... will need to poke around on that a bit more tomorrow...

Reply:
at first glance, looks like it should be possible to run something like `kubectl delete raycluster -l nodelete=true`, but want to test it out before making the code change

Reply:
Yep, works great with a label, so cleanup is as simple as the command in my previous comment!

Reply:
Looks good.  How about putting a debug log when cluster delete is skipped?